builder/dockerfile/parser/parser.go

```diff
@@ -171,11 +171,20 @@ func Parse(rwc io.Reader, d *Directive) (*Node, error) {
 		}
 		startLine := currentLine
 
+		warning := false
 		if line != "" && child == nil {
 			for scanner.Scan() {
 				newline := scanner.Text()
+				// Keep the original content of the current instruction
+				scannedLine += "\n" + newline
```

Comment:
Instead of re-using this variable that was intended to be just an intermediary in a calculation , which also requires both string concatenation and splitting later on, how about storing the lines in a slice and avoiding the concat+split?